app/scripts/views/sign_in.js

```diff
@@ -113,51 +113,30 @@ function (Cocktail, p, BaseView, FormView, SignInTemplate, Session,
         return p.reject(AuthErrors.toError('UNEXPECTED_ERROR'));
       }
 
-      // saved in case there is an error and the
-      // account becomes locked in onSignInError
-      self._signInAccount = account;
-
-      return p().then(function () {
-        if (account.get('password')) {
-          return self.broker.beforeSignIn(account.get('email'))
-            .then(function () {
-              return self.fxaClient.signIn(account.get('email'),
-                      account.get('password'), self.relier);
-            })
-            .then(function (updatedSessionData) {
-              account.set(updatedSessionData);
-              return account;
-            });
-        } else if (account.get('sessionToken')) {
-          // We have a cached Sync session so just check that it hasn't expired.
-          return self.fxaClient.recoveryEmailStatus(account.get('sessionToken'))
-            .then(function (result) {
-              // The result includes the latest verified state
-              account.set(result);
-              return account;
-            });
-        } else {
-          return p.reject(AuthErrors.toError('UNEXPECTED_ERROR'));
-        }
+      return self.broker.beforeSignIn(account.get('email'))
+      .then(function () {
+        return self.user.signInAccount(account, self.relier);
       })
       .then(function (account) {
-        return self.user.setSignedInAccount(account)
-          .then(function () {
-            return account;
+        console.log('account???', account.get('sessionToken'), account);
+        if (self.relier.needsPermissions(account)) {
+          self.navigate('signin_permissions', {
```

Comment:
How you pass the account is slightly different to how its done in [onSignInUnverified](https://github.com/mozilla/fxa-content-server/pull/2377/files#diff-ff872bd2bd254614b59baa704fbfd876R169)